src-self-hosted/translate_c.zig

```diff
@@ -924,6 +1053,327 @@ fn transImplicitCastExpr(
+                .OCLImage2dMSAADepthRO,
+                .OCLImage2dArrayMSAADepthRO,
+                .OCLImage3dRO,
+                .OCLImage1dWO,
+                .OCLImage1dArrayWO,
+                .OCLImage1dBufferWO,
+                .OCLImage2dWO,
+                .OCLImage2dArrayWO,
+                .OCLImage2dDepthWO,
+                .OCLImage2dArrayDepthWO,
+                .OCLImage2dMSAAWO,
+                .OCLImage2dArrayMSAAWO,
+                .OCLImage2dMSAADepthWO,
+                .OCLImage2dArrayMSAADepthWO,
+                .OCLImage3dWO,
+                .OCLImage1dRW,
+                .OCLImage1dArrayRW,
+                .OCLImage1dBufferRW,
+                .OCLImage2dRW,
+                .OCLImage2dArrayRW,
+                .OCLImage2dDepthRW,
+                .OCLImage2dArrayDepthRW,
+                .OCLImage2dMSAARW,
+                .OCLImage2dArrayMSAARW,
+                .OCLImage2dMSAADepthRW,
+                .OCLImage2dArrayMSAADepthRW,
+                .OCLImage3dRW,
+                .OCLSampler,
+                .OCLEvent,
+                .OCLClkEvent,
+                .OCLQueue,
+                .OCLReserveID,
+                .ShortAccum,
+                .Accum,
+                .LongAccum,
+                .UShortAccum,
+                .UAccum,
+                .ULongAccum,
+                .ShortFract,
+                .Fract,
+                .LongFract,
+                .UShortFract,
+                .UFract,
+                .ULongFract,
+                .SatShortAccum,
+                .SatAccum,
+                .SatLongAccum,
+                .SatUShortAccum,
+                .SatUAccum,
+                .SatULongAccum,
+                .SatShortFract,
+                .SatFract,
+                .SatLongFract,
+                .SatUShortFract,
+                .SatUFract,
+                .SatULongFract,
+                .OCLIntelSubgroupAVCMcePayload,
+                .OCLIntelSubgroupAVCImePayload,
+                .OCLIntelSubgroupAVCRefPayload,
+                .OCLIntelSubgroupAVCSicPayload,
+                .OCLIntelSubgroupAVCMceResult,
+                .OCLIntelSubgroupAVCImeResult,
+                .OCLIntelSubgroupAVCRefResult,
+                .OCLIntelSubgroupAVCSicResult,
+                .OCLIntelSubgroupAVCImeResultSingleRefStreamout,
+                .OCLIntelSubgroupAVCImeResultDualRefStreamout,
+                .OCLIntelSubgroupAVCImeSingleRefStreamin,
+                .OCLIntelSubgroupAVCImeDualRefStreamin,
+                => return res.node,
+
+                else => {},
+            }
+        },
+        .Pointer => return transCreateNodeNotEqual(rp, scope, res.node, try transCreateNodeNullLiteral(rp.c)),
+
+        .Typedef => {
+            return transCreateNodeNotEqual(rp, scope, res.node, try transCreateNodeInt(rp.c, 0)); // TODO currently assuming it is like an int/char/bool builtin type. Coerce the type and recurse? Add a toTypedefZeroCmp function?
```

Comment:
Use `ZigClangTypedefNameDecl_getUnderlyingType` here, if then handle it if it is a builtin type.